guidelines.md

```diff
@@ -62,3 +67,10 @@ Using an IP address or some other ephemeral data as part of a device's ID can
 lead to numerous issues. For instance, if the router assigns a new IP address
 to the device, it will no longer be usable via the WebThings UI. Instead,
 something more static should be used.
+
+#### Use Invalid Characters in Device ID
+
+Device IDs are eventually used directly in thing URLs for the gateway. As such,
+using spaces, slashes, etc. in the ID can lead to issues. If your ID is
+generated from user input, you should instead hash the input such that you are
+only using valid characters for a URL path component.
```

Comment:
Just out of interest, why is the advice to hash the input rather than URL encode?

Reply:
Not sure if you've seen [this PR](https://github.com/mozilla-iot/gateway/pull/2288) yet, but URL encoding is not always sufficient, it seems.